applicationset/controllers/clustereventhandler.go

```diff
@@ -82,3 +95,42 @@ func (h *clusterSecretEventHandler) queueRelatedAppGenerators(q addRateLimitingI
 		}
 	}
 }
+
+// nestedGeneratorsHaveClusterGenerator iterate over provided nested generators to check if they have a cluster generator.
+func (h *clusterSecretEventHandler) nestedGeneratorsHaveClusterGenerator(generators []argoprojiov1alpha1.ApplicationSetNestedGenerator) bool {
+	for _, generator := range generators {
+		if h.nestedGeneratorHasClusterGenerator(generator) {
+			return true
+		}
+	}
+	return false
+}
+
+// nestedGeneratorHasClusterGenerator checks if the provided generator has a cluster generator.
+func (h *clusterSecretEventHandler) nestedGeneratorHasClusterGenerator(nested argoprojiov1alpha1.ApplicationSetNestedGenerator) bool {
+	if nested.Clusters != nil {
+		return true
+	}
+
+	if nested.Matrix != nil {
+		nestedMatrix, err := argoprojiov1alpha1.ToNestedMatrixGenerator(nested.Matrix)
+		if err != nil {
+			h.Log.WithError(err).Error("unable to get nested matrix generator")
```

Comment:
Is something like this possible?

```suggestion
			h.Log.WithError(err).Errorf("unable to get nested matrix generator: %w", err)
```

Reply:
You mean, removing the `WithError` ? What's the benefit of this ?

Reply:
I'm a fool. There is no benefit, just missed the WithError. Thanks!

Reply:
Oh! I see, will be fix with resolution of previous comment. Thx.